### PR TITLE
Fix learning rate override log

### PR DESCRIPTION
**Pull request by Maxusmusti**
Logging an incorrect value for overriden learning rate in phased training

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/model/accelerated_train.py [hunk 1]
```diff
@@ -499,7 +499,7 @@ def _training_phase(
 
     if learning_rate is not None:
```
Comment (cdoern) [inline src/instructlab/model/accelerated_train.py:500]: I have one other question, isnt `learning_rate` always not None? if so, won't this always happen

Comment (cdoern) [reply src/instructlab/model/accelerated_train.py:500]: resolved on slack

**Comment by cdoern** [comment]
overriding CI failure, known issue: #2753

**Review by cdoern — APPROVED** [review]

**Comment by cdoern** [comment]
@Mergifyio backport release-v0.21

**Review by khaledsulayman — APPROVED** [review]
